core/src/main/scala/kanaloa/reactive/dispatcher/queue/QueueProcessor.scala

```diff
@@ -13,158 +10,151 @@ import kanaloa.reactive.dispatcher.{Backend, ResultChecker}
-      context become monitoring(newHistory)(pool)
-      onWorkError(newHistory, pool)
-    }
+  def workError(): Unit = {
+    this.resultHistory = resultHistory.enqueueFinite(false, resultHistoryLength)
+    onWorkError(resultHistory, workerPool)
+  }
 
-    {
-      case ScaleTo(newPoolSize, reason) ⇒
-        log.debug(s"Command to scale to $newPoolSize, currently at ${pool.size} due to ${reason.getOrElse("no reason given")}")
-        metricsCollector.send(Metric.PoolSize(newPoolSize))
-        val toPoolSize = Math.max(settings.minPoolSize, Math.min(settings.maxPoolSize, newPoolSize))
-        val diff = toPoolSize - pool.size
-        if (diff > 0)
-          context become monitoring(resultHistory)(pool ++ (1 to diff).map(createWorker))
-        else if (diff < 0)
-          pool.take(-diff).foreach(_ ! Worker.Retire)
-
-      case WorkCompleted(worker, duration) ⇒
-        context become monitoring(resultHistory.enqueueFinite(true, resultHistoryLength))(pool)
-        metricsCollector.send(Metric.ProcessTime(duration))
-        metricsCollector.send(Metric.WorkCompleted)
-
-      case WorkFailed(_) ⇒
-        workError()
-        metricsCollector.send(Metric.WorkFailed)
-
-      case WorkTimedOut(_) ⇒
-        workError()
-        metricsCollector.send(Metric.WorkTimedOut)
-
-      case Terminated(worker) if pool.contains(worker) ⇒
-        removeWorker(pool, worker, monitoring(resultHistory), "Worker removed")
-
-      case Terminated(`queue`) ⇒
-        log.debug(s"Queue ${queue.path} is terminated")
-        self ! Shutdown(retireQueue = false)
-
-      case QueueMaxProcessTimeReached(queue) ⇒
-        log.warning(s"Queue ${queue.path} is still processing after max process time. Shutting Down")
-        self ! Shutdown(retireQueue = true)
-
-      case qs: QueryStatus ⇒ qs reply RunningStatus(pool)
-
-      case Shutdown(reportTo, timeout, retireQueue) ⇒
-        log.info("Commanded to shutdown. Shutting down")
-        if (retireQueue)
-          queue ! Retire(timeout)
-        else //retire from the workers' side
-          pool.foreach(_ ! Worker.Retire)
-
-        delayedMsg(timeout, ShutdownTimeout)
-        context become shuttingDown(pool, reportTo)
-    }: Receive
+  def receive: Receive = {
+    case ScaleTo(newPoolSize, reason) ⇒
+      log.debug(s"Command to scale to $newPoolSize, currently at ${workerPool.size} due to ${reason.getOrElse("no reason given")}")
+      metricsCollector.send(Metric.PoolSize(newPoolSize))
+      val toPoolSize = Math.max(settings.minPoolSize, Math.min(settings.maxPoolSize, newPoolSize))
+      val diff = toPoolSize - workerPool.size
+      if (diff > 0)
+        (1 to diff).foreach { x ⇒
+          createRoutee()
+        }
+      else if (diff < 0)
+        workerPool.take(-diff).foreach(_ ! Worker.Retire)
+
+    case RouteeCreated(routee) ⇒
+      workerPool = workerPool + createWorker(routee)
+
+    case RouteeFailed ⇒ //?
```

Comment:
Right now, the Future which failed to create the Routee is doing logging.  Anything else we want to do here?